flink-connector/src/main/java/org/apache/gravitino/flink/connector/catalog/BaseCatalog.java

```diff
@@ -470,6 +535,122 @@ private Column toGravitinoColumn(org.apache.flink.table.catalog.Column column) {
         null);
   }
 
+  private List<TableChange> optionsChanges(
+      Map<String, String> currentOptions, Map<String, String> updatedOptions) {
+    List<TableChange> optionsChanges = com.google.common.collect.Lists.newArrayList();
```

Comment:
Gravitino may place some key like `Gravitino id` in table properties, I'm afraid this will remove the key.

Reply:
I removed the option change in this function to provide it. The `alterTable(ObjectPath tablePath, CatalogBaseTable newTable, boolean ignoreIfNotExists) ` must be implemented. But for this function, it is hard to get the TableChange.
```
void alterTable(ObjectPath tablePath, CatalogBaseTable newTable, boolean ignoreIfNotExists)

default void alterTable(
            ObjectPath tablePath,
            CatalogBaseTable newTable,
            List<TableChange> tableChanges,
            boolean ignoreIfNotExists)
            throws TableNotExistException, CatalogException {
        alterTable(tablePath, newTable, ignoreIfNotExists);
    }
```

Reply:
ok, if there are real problems, let's check how to fix it